clients/include/unit.h

```diff
@@ -63,4 +63,21 @@ void unit_check_nrm2(T cpu_result, T gpu_result, T tolerance)
 #endif
 }
 
+template <typename T>
+void unit_check_nrm2(int            batch_count,
+                     host_vector<T> cpu_result,
+                     host_vector<T> gpu_result,
+                     T              tolerance)
```

Comment:
Why not rename the last arg vector_length and just pass in vector length N?  Tolerance implies something to be judged which seems large as 2N.   You can type 2*N once, as later you can also tighten in one place, sqrt(2N),... 2sqrt(N) ?

Reply:
Renamed to vector_length and am now passing in N. Also removed scalar 2 as it looks like all tests pass without it (on cuda and hip).

Reply:
Very good it is already paying off ;-)